pkg/controller/plan/validation.go

```diff
@@ -147,16 +147,15 @@ func (r *Reconciler) validate(plan *api.Plan) error {
 	}
 
 	// Validate version only if migration is OCP to OCP
-	err = r.validateOCPVersion(plan)
-	if err != nil {
+	if err := r.validateOpenShiftVersion(plan); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (r *Reconciler) validateOCPVersion(plan *api.Plan) error {
-	if plan.IsSourceProviderOCP() && plan.Provider.Destination.Type() == api.OpenShift {
+func (r *Reconciler) validateOpenShiftVersion(plan *api.Plan) error {
+	if !plan.Status.HasCondition(validation.SourceProviderNotValid) && plan.IsSourceProviderOCP() && plan.Provider.Destination.Type() == api.OpenShift {
```

Comment:
This happens because we reference a provider that doesn't exist, why is it specific to ocp?
I think we shouldn't even reach here in this case

Reply:
you're right but this is the only verification that refers to the source provider

Reply:
it doesn't matter much but I've extracted that check to the `validate` function

Reply:
actually, changed it again to align this validation functions with the other ones that don't face this issue...